ReSharper.FSharp/src/FSharp.Fantomas.Host/src/FantomasAssemblyResolver.cs

```diff
@@ -21,23 +25,33 @@ static FantomasAssemblyResolver()
       }
     }
 
-    public static Assembly Resolve(object sender, ResolveEventArgs eventArgs)
+    private static Assembly LoadAssembly(IEnumerable<string> paths, string assemblyName)
     {
-      var assemblyName = $"{new AssemblyName(eventArgs.Name).Name}.dll";
-
-      foreach (var path in OurAdditionalProbingPaths)
+      foreach (var path in paths)
       {
         var assemblyPath = Path.Combine(path, assemblyName);
         if (!File.Exists(assemblyPath)) continue;
-
-        var assembly = Assembly.LoadFrom(assemblyPath);
-        return assembly;
+        return Assembly.LoadFrom(assemblyPath);
       }
 
-      Console.Error.Write($"\nFailed to resolve assembly by name '{eventArgs.Name}'" +
-                          $"\n  Requesting assembly: {eventArgs.RequestingAssembly?.FullName}" +
-                          $"\n  Probing paths: {string.Join("\n", OurAdditionalProbingPaths)}");
       return null;
     }
+
+    public static Assembly Resolve(object sender, ResolveEventArgs eventArgs)
+    {
+      var assemblyName = $"{new AssemblyName(eventArgs.Name).Name}.dll";
+      var paths =
+        assemblyName == "Fantomas.dll" || assemblyName == "FSharp.Compiler.Service.dll"
+          ? (IEnumerable<string>) FSharpFantomasAssembliesPath
+          : OurAdditionalProbingPaths;
```

Comment:
Just a thought: what if you just prepend Fantomas-related paths, so it searches them first? Will it work?

Reply:
Do you want to add it to the top of the OurAdditionalProbingPaths list? If Fantomas' FCS assembly is suddenly not found, the search will continue in OurAdditionalProbingPaths and the wrong version will be loaded (custom version from F# plugin).

Reply:
If it's suddenly not found we're already in the trouble anyway, no?

Reply:
> Do you want to add it to the top of the OurAdditionalProbingPaths list?

Yup, so it'd effectively revert to previous logic, it seemed a bit less complicated. Not a big issue, though.